pkg/operator/genericoperatorclient/test_operator_client.go

```diff
@@ -0,0 +1,35 @@
+package genericoperatorclient
+
+import (
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/rest"
+	"net/http"
+)
+
+func NewOperatorClientWithClient(httpClient *http.Client, gvr schema.GroupVersionResource, gvk schema.GroupVersionKind, extractApplySpec OperatorSpecExtractorFunc, extractApplyStatus OperatorStatusExtractorFunc) (v1helpers.OperatorClientWithFinalizers, dynamicinformer.DynamicSharedInformerFactory, error) {
```

Comment:
The file name indicates that the method is intended to be used by test code. However, the package name and the file name itself will expose these methods to the production code. Is that intentional?